<a href="https://colab.research.google.com/github/GyaviWalls/extra_datasets/blob/main/Project_2_Part_1_(Core).ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# Project 2 - Part 1 (Core).ipynb

> Author: Gyavira Ayebare.B.

In [ ]:
import pandas as pd

## First choice: Car Insurance Data


In [ ]:
filename = '/content/drive/MyDrive/CodingDojo/02-IntroML/Week07/Data/Car_Insurance_Claim.csv'
df1 = pd.read_csv(filename)
df1 = df1.set_index('ID')
df1.head()

AGE  GENDER      RACE DRIVING_EXPERIENCE    EDUCATION  \
ID                                                                
569520    65+  female  majority               0-9y  high school   
750365  16-25    male  majority               0-9y         none   
199901  16-25  female  majority               0-9y  high school   
478866  16-25    male  majority               0-9y   university   
731664  26-39    male  majority             10-19y         none   

               INCOME  CREDIT_SCORE  VEHICLE_OWNERSHIP VEHICLE_YEAR  MARRIED  \
ID                                                                             
569520    upper class      0.629027                1.0   after 2015      0.0   
750365        poverty      0.357757                0.0  before 2015      0.0   
199901  working class      0.493146                1.0  before 2015      0.0   
478866  working class      0.206013                1.0  before 2015      0.0   
731664  working class      0.388366                1.0  before 2015      0.0   

        CHILDREN  POSTAL_CODE  ANNUAL_MILEAGE VEHICLE_TYPE  \
ID                                                           
569520       1.0        10238         12000.0        sedan   
750365       0.0        10238         16000.0        sedan   
199901       0.0        10238         11000.0        sedan   
478866       1.0        32765         11000.0        sedan   
731664       0.0        32765         12000.0        sedan   

        SPEEDING_VIOLATIONS  DUIS  PAST_ACCIDENTS  OUTCOME  
ID                                                          
569520                    0     0               0      0.0  
750365                    0     0               0      1.0  
199901                    0     0               0      0.0  
478866                    0     0               0      0.0  
731664                    2     0               1      1.0

In [ ]:
df1.describe()

CREDIT_SCORE  VEHICLE_OWNERSHIP       MARRIED      CHILDREN  \
count   9018.000000       10000.000000  10000.000000  10000.000000   
mean       0.515813           0.697000      0.498200      0.688800   
std        0.137688           0.459578      0.500022      0.463008   
min        0.053358           0.000000      0.000000      0.000000   
25%        0.417191           0.000000      0.000000      0.000000   
50%        0.525033           1.000000      0.000000      1.000000   
75%        0.618312           1.000000      1.000000      1.000000   
max        0.960819           1.000000      1.000000      1.000000   

        POSTAL_CODE  ANNUAL_MILEAGE  SPEEDING_VIOLATIONS         DUIS  \
count  10000.000000     9043.000000         10000.000000  10000.00000   
mean   19864.548400    11697.003207             1.482900      0.23920   
std    18915.613855     2818.434528             2.241966      0.55499   
min    10238.000000     2000.000000             0.000000      0.00000   
25%    10238.000000    10000.000000             0.000000      0.00000   
50%    10238.000000    12000.000000             0.000000      0.00000   
75%    32765.000000    14000.000000             2.000000      0.00000   
max    92101.000000    22000.000000            22.000000      6.00000   

       PAST_ACCIDENTS       OUTCOME  
count    10000.000000  10000.000000  
mean         1.056300      0.313300  
std          1.652454      0.463858  
min          0.000000      0.000000  
25%          0.000000      0.000000  
50%          0.000000      0.000000  
75%          2.000000      1.000000  
max         15.000000      1.000000

In [ ]:
df1.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 10000 entries, 569520 to 442696
Data columns (total 18 columns):
 #   Column               Non-Null Count  Dtype  
---  ------               --------------  -----  
 0   AGE                  10000 non-null  object 
 1   GENDER               10000 non-null  object 
 2   RACE                 10000 non-null  object 
 3   DRIVING_EXPERIENCE   10000 non-null  object 
 4   EDUCATION            10000 non-null  object 
 5   INCOME               10000 non-null  object 
 6   CREDIT_SCORE         9018 non-null   float64
 7   VEHICLE_OWNERSHIP    10000 non-null  float64
 8   VEHICLE_YEAR         10000 non-null  object 
 9   MARRIED              10000 non-null  float64
 10  CHILDREN             10000 non-null  float64
 11  POSTAL_CODE          10000 non-null  int64  
 12  ANNUAL_MILEAGE       9043 non-null   float64
 13  VEHICLE_TYPE         10000 non-null  object 
 14  SPEEDING_VIOLATIONS  10000 non-null  int64  
 15  DUIS                 10000 non

1. Source of data
> Obtained from the kaggle site [car insurance dataset](https://www.kaggle.com/datasets/sagnik1511/car-insurance-data)

2. Brief description of data
> It continas annual car insurance data. Represented as features that can be used to find out the real customer behaviors over the data, and also predict whether they shall claim their insurance or not.



3. What is the target?
> The outcome feature which shows 1, if a customer has claimed his/her loan, else 0.

4. What does one row represent? (A person? A business? An event? A product?)
> The claim details provided to the insurance company by the car owner

5. Is this a classification or regression problem?
> Classification problem

6. How many features does the data have?
> 18 features

7. How many rows are in the dataset?
> 10000 rows

8. What, if any, challenges do you foresee in cleaning, exploring, or modeling this dataset?

>
* The target will have to be converted into a binary output for it seems to be in a float.
* imputing some null values for features CREDIT_SCORE and ANNUAL_MILEAGE
* plus there might be some extra type conversions.

## Second choice: Adult income dataset

In [ ]:
filename = '/content/drive/MyDrive/CodingDojo/02-IntroML/Week07/Data/adult.csv'
df2 = pd.read_csv(filename)
# df2 = df2.set_index('ID')
df2.head()

age  workclass  fnlwgt     education  educational-num      marital-status  \
0   25    Private  226802          11th                7       Never-married   
1   38    Private   89814       HS-grad                9  Married-civ-spouse   
2   28  Local-gov  336951    Assoc-acdm               12  Married-civ-spouse   
3   44    Private  160323  Some-college               10  Married-civ-spouse   
4   18          ?  103497  Some-college               10       Never-married   

          occupation relationship   race  gender  capital-gain  capital-loss  \
0  Machine-op-inspct    Own-child  Black    Male             0             0   
1    Farming-fishing      Husband  White    Male             0             0   
2    Protective-serv      Husband  White    Male             0             0   
3  Machine-op-inspct      Husband  Black    Male          7688             0   
4                  ?    Own-child  White  Female             0             0   

   hours-per-week native-country income  
0              40  United-States  <=50K  
1              50  United-States  <=50K  
2              40  United-States   >50K  
3              40  United-States   >50K  
4              30  United-States  <=50K

In [ ]:
df2.describe()

age        fnlwgt  educational-num  capital-gain  \
count  48842.000000  4.884200e+04     48842.000000  48842.000000   
mean      38.643585  1.896641e+05        10.078089   1079.067626   
std       13.710510  1.056040e+05         2.570973   7452.019058   
min       17.000000  1.228500e+04         1.000000      0.000000   
25%       28.000000  1.175505e+05         9.000000      0.000000   
50%       37.000000  1.781445e+05        10.000000      0.000000   
75%       48.000000  2.376420e+05        12.000000      0.000000   
max       90.000000  1.490400e+06        16.000000  99999.000000   

       capital-loss  hours-per-week  
count  48842.000000    48842.000000  
mean      87.502314       40.422382  
std      403.004552       12.391444  
min        0.000000        1.000000  
25%        0.000000       40.000000  
50%        0.000000       40.000000  
75%        0.000000       45.000000  
max     4356.000000       99.000000

In [ ]:
df2.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 48842 entries, 0 to 48841
Data columns (total 15 columns):
 #   Column           Non-Null Count  Dtype 
---  ------           --------------  ----- 
 0   age              48842 non-null  int64 
 1   workclass        48842 non-null  object
 2   fnlwgt           48842 non-null  int64 
 3   education        48842 non-null  object
 4   educational-num  48842 non-null  int64 
 5   marital-status   48842 non-null  object
 6   occupation       48842 non-null  object
 7   relationship     48842 non-null  object
 8   race             48842 non-null  object
 9   gender           48842 non-null  object
 10  capital-gain     48842 non-null  int64 
 11  capital-loss     48842 non-null  int64 
 12  hours-per-week   48842 non-null  int64 
 13  native-country   48842 non-null  object
 14  income           48842 non-null  object
dtypes: int64(6), object(9)
memory usage: 5.6+ MB


1. Source of data
> Obtained from the kaggle site [adult income dataset](https://www.kaggle.com/datasets/wenruliu/adult-income-dataset)

2. Brief description of data
> The data contains an individual’s annual income results from various factors. These are influenced by the individual’s education level, age, gender, occupation, and etc.
 With this, the possibility in predicting income level based on the individual’s personal information can be achieved.

3. What is the target?
> The Income feature judged based on whether an individual earns less than or equal to 50K or greater than 50K

4. What does one row represent? (A person? A business? An event? A product?)
> The data of a person

5. Is this a classification or regression problem?
> Classification problem

6. How many features does the data have?
> 15 features

7. How many rows are in the dataset?
> 48842 rows

8. What, if any, challenges do you foresee in cleaning, exploring, or modeling this dataset?
>
* I might need to drop the education and occupation features, high cardinality issues
* Also converting the target feature to the appropriate class